### PR TITLE
Sprint4

### PR DESCRIPTION
**Pull request by cbizon**
Updated with discussion from TACT call on 9/27

### CODE REVIEW

**Comment by tursynay** [comment]
all, you have been assigned to review the Spring 4 PR, please take a look at the file and complete your review so Chris merges this PR. Please complete no later than eob Monday July 1

**Review by andrewsu — APPROVED** [review]

**Review by sierra-moxon — APPROVED** [review]
Per our conversation yesterday in DM call and today at TAQA, we should consider "removing redundant Chembl 'in clinical trials' ingests as a priority next sprint"

**Comment by gglusman** [comment]
> Per our conversation yesterday in DM call and today at TAQA, we should consider "removing redundant Chembl 'in clinical trials' ingests as a priority next sprint"

Completely agree. There was widespread support for improving how we handle clinical trials info at the relay, then somehow that morphed into "it's not a priority, and we're not doing it". Please, let's reconsider this!

Edited to add: On first read, I assumed Sierra's "next sprint" refers to the one we're starting. To disambiguate, I don't think this should wait until Sprint 5.
I added an outline of tasks for Sprints 4 through 7 in the Sprint Planning table.

**Review by newgene — APPROVED** [review]

**Review by maximusunc — APPROVED** [review]

**Review by karafecho — APPROVED** [review]

**Review by vdancik — APPROVED** [review]

**Comment by webyrd** [comment]
I think I'm okay with everything, with the possible exception of this:

# Fix Failing Automated Tests
## Fugu (CI):
- Each ARA is expected to pass 40 tests from the Sprint 4 test suite, which focuses on TopAnswer results.

As I understand it, the idea of TopAnswer results is to ensure we can generate ground-truth results.  Unsecret is focusing on creative results rather than ground truth, and boosts the scores of creative answers.  As a result, we will return ground truth answers less often than we might.  This is by design, since we are responding to "creative mode" queries.

**Review by GregHydeDartmouth — APPROVED** [review]

**Review by saramsey — APPROVED** [review]

File: PrioritiesSprint5.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- Run MCQ Suite
+
+# Fix Failing Automated Tests
+## Guppy (CI):
+- Each ARA is expected to pass X tests from the Sprint 5 test suite
+
+# Collect Feedback Issues
+## Fugu (TEST):
+- Each team must perform manual testing of the Fugu release in TEST and report issues to the Feedback repo
+between July 2 and July 9, 2024
```
Comment (CaseyTa) [inline PrioritiesSprint5.md:55]: Was this supposed to be Aug 2-9?

**Comment by CaseyTa** [comment]
> I think I'm okay with everything, with the possible exception of this:
> 
> # Fix Failing Automated Tests
> ## Fugu (CI):
> * Each ARA is expected to pass 40 tests from the Sprint 4 test suite, which focuses on TopAnswer results.
> 
> As I understand it, the idea of TopAnswer results is to ensure we can generate ground-truth results. Unsecret is focusing on creative results rather than ground truth, and boosts the scores of creative answers. As a result, we will return ground truth answers less often than we might. This is by design, since we are responding to "creative mode" queries.

Would it be allowed for different ARAs to develop their own test suites and passing criteria that they think would be representative to their philosophy?

**Review by CaseyTa — APPROVED** [review]

**Comment by webyrd** [comment]
> > I think I'm okay with everything, with the possible exception of this:
> > # Fix Failing Automated Tests
> > ## Fugu (CI):
> > 
> > * Each ARA is expected to pass 40 tests from the Sprint 4 test suite, which focuses on TopAnswer results.
> > 
> > As I understand it, the idea of TopAnswer results is to ensure we can generate ground-truth results. Unsecret is focusing on creative results rather than ground truth, and boosts the scores of creative answers. As a result, we will return ground truth answers less often than we might. This is by design, since we are responding to "creative mode" queries.
> 
> Would it be allowed for different ARAs to develop their own test suites and passing criteria that they think would be representative to their philosophy?

Good question.

I'll approve the changes, with this single reservation that I think would be good to discuss.

**Review by webyrd — APPROVED** [review]
I approve the changes, with a single concern:

# Fix Failing Automated Tests
## Fugu (CI):
- Each ARA is expected to pass 40 tests from the Sprint 4 test suite, which focuses on TopAnswer results.

As I understand it, the idea of TopAnswer results is to ensure we can generate ground-truth results.  Unsecret is focusing on creative results rather than ground truth, and boosts the scores of creative answers.  As a result, we will return ground truth answers less often than we might.  This is by design, since we are responding to "creative mode" queries.

@CaseyTa asked "Would it be allowed for different ARAs to develop their own test suites and passing criteria that they think would be representative to their philosophy?"

I think this is a reasonable question that would be good to discuss.

File: PrioritiesSprint4.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+## Pre-development work:
+- Experiment with and refine the summarization idea: Decide whether we will implement single-answer summarization, all-answers summarization, user querying over answers (RAG-like), or other approaches.
+
+# O&O Based grouping in the UI 
+## Pre-development work:
+- User testing to determine effectiveness of chemical grouping based on ATC and Chebi roles
+- User testing to determine effectiveness of gene grouping based on Genetics KP Gene Sets and PFOCR
+## Fugu (CI):
+- Implementation of Gene Set grouping API (Genetics KP)
+
```
Comment (sharatisrani) [inline PrioritiesSprint4.md:41]: Implementation of TDL in Annotator APIs

Comment (sharatisrani) [inline PrioritiesSprint4.md:41]: Implementation of TDL in Annotator APIs

**Review by suihuang-ISB — APPROVED** [review]

File: PrioritiesSprint4.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+# LLM Summarization 
+## Pre-development work:
+- Experiment with and refine the summarization idea: Decide whether we will implement single-answer summarization, all-answers summarization, user querying over answers (RAG-like), or other approaches.
+
+# O&O Based grouping in the UI 
+## Pre-development work:
+- User testing to determine effectiveness of chemical grouping based on ATC and Chebi roles
+- User testing to determine effectiveness of gene grouping based on Genetics KP Gene Sets and PFOCR
+## Fugu (CI):
```
Comment (sstemann) [inline PrioritiesSprint4.md:39]: Per Marc Duby on 7/2, - Implementation of Gene Set grouping API (Genetics KP) milestone will be in their Dev environment by 7/19 but not in CI.

**Pull request merged**